src/assets/scripts/client/info/AirportInfoController.js

```diff
@@ -0,0 +1,287 @@
+     * @for AirportInfoController
+     * @method _setupHandlers
+     * @chainable
+     * @private
+     */
+    _setupHandlers() {
+        this.onAirportChangeHandler = this.onAirportChange.bind(this);
+
+        return this;
+    }
+
+    /**
+     * Set initial element references
+     *
+     * Should be run once only on instantiation
+     *
+     * @for StripViewModel
+     * @method _createChildren
+     * @chainable
+     * @private
+     */
+    _createChildren() {
+        this.$element.append(this.$template);
+
+        return this;
+    }
+
+    /**
+     * Enable all event handlers
+     *
+     * @for AirportInfoController
+     * @method _enable
+     * @chainable
+     * @private
+     */
+    _enable() {
+        this._eventBus.on(EVENT.AIRPORT_CHANGE, this.onAirportChangeHandler);
+        this.onAirportChange();
+
+        return this;
+    }
+
+    /**
+     * Disable all event handlers and destroy the instance
+     *
+     * @for AirportInfoController
+     * @method _disable
+     * @chainable
+     * @private
+     */
+    _disable() {
+        this.$element = null;
+        this.$template = null;
+        this.simClockController = null;
+        this.airportInfoController = null;
+
+        this._eventBus.off(EVENT.AIRPORT_CHANGE, this._onAirportChangeHandler);
+
+        return this;
+    }
+
+    /**
+     * @for AirportInfoController
+     * @method reset
+     * @chainable
+     */
+    reset() {
+        this.wind = null;
+        this.altimeter = null;
+        this.elevation = null;
+        this.icao = null;
+
+        return this;
+    }
+
+    /**
+     * Updates the information taken from the AirportModel: the wind, the altimeter,
+     * and the elevation. Triggered on airport change.
+     *
+     * @for AirportInfoController
+     * @method onAirportChange
+     */
+    onAirportChange() {
+        const airport = AirportController.airport_get();
+        const icao = airport.icao.toUpperCase();
+        const windAngle = Math.round(radiansToDegrees(airport.wind.angle));
+
+        this.wind = this._buildWindAndGustReadout({ speed: airport.wind.speed, angle: windAngle });
+        this.altimeter = this._generateAltimeterReading(airport.wind.speed);
+        this.elevation = `${airport.elevation}`;
+        this.icao = icao;
+
+        this._render();
+    }
+
+    /**
+     * Sets the values from the updated airport info.
+     *
+     * @for AirportInfoController
+     * @method _render
+     * @private
+     */
+    _render() {
+        this.$windView.text(`${this.icao} ${this.wind}`);
+        this.$altimeterView.text(`${this.icao} ${this.altimeter}`);
+        this.$elevationView.text(`${this.icao} ${this.elevation}`);
+    }
+
+    /**
+     * Updates the clock, called from `AppController#update_pre`
+     *
+     * @for AirportInfoController
+     * @method updateClock
+     */
+    updateClock() {
+        this.simClockController.update();
+
+        const time = this.simClockController.render();
```

Comment:
whaa?  

1) why does `.render()` have a return value?
2) why are we using that to set _another_ `$clockView` here?

I think I know where you're going with this, which is ok, but the naming is a little confusing.

Reply:
1. It returns the formatted time.
2. Would do you mean by setting **another** `$clockView`?

`simClockController.render()` only formats the clock string. It does not change the on-screen value. #notmywork 😉 

I have that feeling that you are going to tell me that `simClockController.render` is the right place to call `$clockView.text(time)`.

Reply:
done